### ISSUE TITLE
Installation Error

### ISSUE THREAD

**Issue by iPFAS** [GT4SD/gt4sd-core#216]
**Describe the bug**
    I tried to install the GPU version, but there were many software package dependency conflicts. Even after I manually installed it, the problem still could not be solved.For example, the following pip command.
    There are many such problems that prevent me from using such an excellent framework for scientific research, which is regrettable.

**To Reproduce**

1. pip install  git+https://github.com/PaccMann/paccmann_generator@0.0.2 --user
![QQ截图20230425193100](https://user-images.githubusercontent.com/114653967/234263617-b7b15d58-eb85-4296-8436-709856b3ceaa.png)

**Comment by jannisborn** [comment]
Hi @iPFAS,
Thanks for getting in touch. Can you please report the commands that you executed? We have a thorough test suite for Ubuntu and most of the developers use Mac.

Note that generally, you dont have to install any packages manually, just follow the installation workflow as described in the README

**Comment by jannisborn** [comment]
Ok, thx for sharing those details. Let's make a step back: after activating the env, it seems that you did not `pip install gt4sd` or `pip install -e .`? This would clearly be needed in the first place before you can use any GT4SD (including the trainer)

**Comment by iPFAS** [comment]
Sorry for the late reply. The main reason is that the installation process for reproducing the issue takes too long. There is a detailed explanation below. Of course, I didn’t miss this step. In fact I’ve tried to install it many times before, but I couldn’t get it done and that’s why I submitted an issue. 

I replaced it with another CentOS Stream server that has a cleaner installation environment to reproduce my issue. The process is as follows.

```
    sudo conda env create -f conda_gpu.yml
    sudo conda activate gt4sd
    pip --no-cache-dir install gt4sd
    pip install -r vcs_requirements.txt
```

- sudo conda env create -f conda_gpu.yml ---- executed successfully without any issues.
![1](https://user-images.githubusercontent.com/114653967/234344145-80001a65-0f9d-443c-b429-cdc515da161b.jpg)
![2](https://user-images.githubusercontent.com/114653967/234344247-30850fe5-32f9-4fcb-9c38-f099946a791c.jpg)
- pip --no-cache-dir install gt4sd ---- raise an error ：ImportError: PyTorch is required to install pytorch-fast-transformers. Please install your favorite version of PyTorch, we support 1.3.1, 1.5.0 and >=1.6
  The installation log shows that PyTorch was installed. I’m not sure why there is an error here. Perhaps only torch was installed but not PyTorch, which caused this?
![3](https://user-images.githubusercontent.com/114653967/234344313-c89d4acb-6411-4dab-88b3-dcf2f1b0f9bd.jpg)
_Collecting torch
Downloading torch-1.13.1-cp37-cp37m-manylinux1_x86_64.whl (887.5 MB)_
- This installation step requires a lot of time because pip needs to search many versions of the package.
![1](https://user-images.githubusercontent.com/114653967/234460503-2d25553c-8f2d-4657-868a-96e291151c4e.png)
- another error
![2](https://user-images.githubusercontent.com/114653967/234491903-d841a6df-5b40-49bb-8c9d-88b5b046ed1b.png)
- pip install -r vcs_requirements.txt ---- There is still an error caused by version dependencies here
![2](https://user-images.githubusercontent.com/114653967/234493396-83cedd02-40ad-4548-8cce-8cda4249399a.png)
Of course, these issues are relatively easy to solve, so I upgraded them one by one according to the prompts
```
pip install scikit-learn==0.23.2
pip install importlib-resources==5.10.0
pip install numpy==1.21
```
- then, new errors about version dependencies occured
![image](https://user-images.githubusercontent.com/114653967/234494253-e4853247-bcb0-431e-9e94-86b4c829a738.png)
and the new requirement error of numpy : yellowbrick 1.3.post1 requires numpy<1.20,>=1.16.0, but you'll have numpy 1.21.0 which is incompatible.
that means ：after I upgraded numpy, another package prompted me that the version I upgraded exceeded the requirements, which cannot be solved. 
Of course, sometimes the version issue of numpy can be ignored and does not affect the use of the program. I ignored it and continued.
- So I continued to upgrade the versions according to the prompts
```
pip install protobuf==3.19.6
pip install scikit-learn==1.0.0
```
- There are still two errors left. I decided to ignore them and test using gt4sd-inference --help or gt4sd-trainer --help
![3](https://user-images.githubusercontent.com/114653967/234498736-cac8ae57-ba32-4d1b-bd93-e2c82fb97aac.png)
```
gt4sd-inference --help
````
However, after experiencing so many difficulties and obstacles, I still cannot run it successfully. The prompt of Segmentation fault (core dumped) makes me heartbroken. Looking at the error message, it is probably a problem with tensorflow. At this point, I may have to solve the tensorflow environment problem again. 
![image](https://user-images.githubusercontent.com/114653967/234499334-abada57c-a558-4d41-a536-69c0a5892e9e.png)
Of course, this is not your problem. Your framework is excellent. However, if the installation is not simple enough, the pre-investment cost of using this framework will be too high, which may cause us researchers to seek other frameworks

**Comment by jannisborn** [comment]
Hi @iPFAS,

Sorry that you are experience such issues! Let's go step by step.
1. `sudo conda env create -f conda_gpu.yml` --> works
2. `pip --no-cache-dir install gt4sd` --> This is a red flag. VERY surprising that `pytorch` seems to not be installed, especially since the previous command was successful. If you check the `conda_gpu.yml` file you'll see that it should take care of the `pytorch` installation. Did you manually verify that `pytorch` was indeed not installed after step 1?
3. `pip install -r vcs_requirements.txt` --> This should not be necessary since it's being executed from inside the `conda_gpu.yml`. Generally these version mismatch log messages are more **warnings** than actual **errors**, this is just `pip` telling you about potential problem sources. However, when I set up the env with the current tip of master, I do not see such warnings.
4. `gt4sd-inference --help` --> Thx for reporting the segfault! It's interesting, I've seen this issue in the past it's caused by pytorch-lightning. It's a common issue described here: https://github.com/Lightning-AI/lightning/issues/11663 We thought that we had mitigated that by fixing the relation with `sentencepice`. Can you check whether you get a segfault if you open python and just import pytorch lightning? Probably yes. Probably, those parts of the library that do not import lightning are functional. That's obviously not a permanent solution. We're looking into this currently and try to fix it.

So overall, from all the things you report only 4) is something that we can reproduce. Which version of `conda` do you have? We use `4.12.0`, you might experience issues with much more recent versions

**Comment by iPFAS** [comment]
Thank you for your help. I followed your advice and downgraded conda to 4.12.0 and tried again. Unfortunately, I still cannot install it successfully. Overall, it is probably due to my network and environment. Considering the requirements of work progress, I can only try again later when I have time. Of course, thank you very much for your timely and professional reply.I will close the issue Thank you again.

**Comment by jannisborn** [comment]
Hi @iPFAS,

We just released a new version (1.3.1) which fixes some issues that multiple users observed in the installation/setup including the segmentation fault you reported. Please feel free to retry the installation with the latest version!
Thanks for your patience

**Comment by iPFAS** [comment]
Hi，
Thank you for letting me know about the new version and the fixes it includes. I appreciate your team’s efforts in addressing the issues and will definitely consider retrying the installation with the latest version. Thanks for your support
